ionoscloud/data_source_ipfailover.go

```diff
@@ -44,17 +46,45 @@ func dataSourceIpFailover() *schema.Resource {
 }
 
 func dataSourceIpFailoverRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CloudApiClient
+	dcId := d.Get("datacenter_id").(string)
+	lanId := d.Get("lan_id").(string)
+	ip := d.Get("ip").(string)
 
-	id, idOk := d.GetOk("id")
-
-	if !idOk {
-		return diag.FromErr(fmt.Errorf("please provide the ip failover id"))
+	lan, apiResponse, err := client.LANsApi.DatacentersLansFindById(ctx, dcId, lanId).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return diag.FromErr(fmt.Errorf("unable to find the LAN with ID: %s, datacenter ID: %s", lanId, dcId))
+		}
+		return diag.FromErr(fmt.Errorf("error while fetching LAN with ID: %s, datacenter ID: %s", lanId, dcId))
+	}
+	if lan.Properties == nil || lan.Properties.IpFailover == nil {
+		return diag.FromErr(fmt.Errorf("expected a LAN response containing IP failover groups but received 'nil' instead"))
 	}
-	d.SetId(id.(string))
 
-	if diags := resourceLanIPFailoverRead(ctx, d, meta); diags != nil {
-		return diags
+	ipFailoverGroups := lan.Properties.IpFailover
+	ipFailoverGroupFound := false
+	if lan.Properties != nil && ipFailoverGroups != nil && len(*ipFailoverGroups) > 0 {
+		for _, ipFailoverGroup := range *ipFailoverGroups {
+			// Search for the appropiate IP Failover Group using the provided IP
+			if *ipFailoverGroup.Ip == ip {
+				// Set the information only if the IP Failover Group exists
+				d.SetId(uuidgen.ResourceUuid().String())
```

Comment:
I think we can move setting things to after the if !ipFailoverGroupFound { } below to keep this leaner, but as you wish

Reply:
@cristiGuranIonos I will leave this as it is, I like it this way.